table/tables/tables.go

```diff
@@ -474,6 +474,11 @@ func (t *tableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, skipHan
 			if err != nil {
 				return 0, errors.Trace(err)
 			}
+			if col.Offset < len(r) {
```

Comment:
You can call `ctx.StmtAddDirtyTableOP(table.DirtyTableAddRow, t.physicalTableID, recordID, row)` in line 508